numpy/lib/polynomial.py

```diff
@@ -441,6 +441,12 @@ def polyfit(x, y, deg, rcond=None, full=False, w=None, cov=False):
     method is recommended for new code as it is more stable numerically. See
     the documentation of the method for more information.
 
+    Do not confuse this function with the very similar
```

Comment:
I think I'd be happy to be more explicit that the other function is preferred, i.e., something like:
"Note that we generally recommend to use the very similar..."
(I also like your suggestion of a deprecation warning, but think it is better to take it a step at the time, as you do here!)